mbedtls/mbedtls.py

```diff
@@ -41,6 +31,15 @@ def harness(dut: Dut, ctx: TestContext, result: TestResult):
             # if there are dots after test name - remove them
             subname = re.sub(r" \.+ ", "", parsed["name"])
             subresult = TestSubResult(subname=subname, status=status)
+
+            if msg and subresult.status == Status.FAIL:
```

Comment:
I'm not sure but this would result in appending extra message to invalid testcase (the next one) - assuming the format is:
```
[test name].... FAIL
   [extra messages why it failed]
[test name 2].... SUCCESS
```


IMHO this could be simplified by not resetting `subresult` variable and appending to it directly, eg.

```python
    while True:
        idx = dut.expect([RESULT, FINAL, MSG_LINE], timeout=90)
        parsed = dut.match.groupdict()

        if idx == 2:
            # append extra message to the last test (if available)
            if subresult:
                subresult.msg += parsed["line"] + " "

        if idx == 0:
            # the subresult finished running - add it to result now to ensure correct run time
            if parsed["status"] == "----":
                status = Status.SKIP
            else:
                status = Status.from_str(parsed["status"])

            if status == Status.FAIL:
                test_status = Status.FAIL

            # if there are dots after test name - remove them
            subname = re.sub(r" \.+ ", "", parsed["name"])
            subresult = TestSubResult(subname=subname, status=status)
            result.add_subresult_obj(subresult)

        if idx == 1:
            break

    return TestResult(status=test_status)
```